dev/io.openliberty.microprofile.openapi.internal.common/src/io/openliberty/microprofile/openapi/internal/common/OpenAPIAppConfigProviderImpl.java

```diff
@@ -0,0 +1,255 @@
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(this, tc, "Initial processing of server.xml");
+            }
+
+            MpConfigServerConfigObject newConfig = new MpConfigServerConfigObject(properties);
+            config = newConfig;
+        }
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> properties) {
+        if (ProductInfo.getBetaEdition()) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(this, tc, "Processing update to server.xml");
+            }
+
+            MpConfigServerConfigObject newConfig = new MpConfigServerConfigObject(properties);
+
+            if (config.equals(newConfig)) {
+                Tr.event(this, tc, "After update there is no need to rebuild the openAPI document");
+            } else {
+                Tr.event(this, tc, "Clearing out the outdated openAPI document");
+
+                config = newConfig;
+                openAPIAppConfigListeners.sort((OpenAPIAppConfigListener o1, OpenAPIAppConfigListener o2) -> Integer.compare(o1.getConfigListenerPriority(),
+                                                                                                                             o2.getConfigListenerPriority()));
+
+                for (OpenAPIAppConfigListener listener : openAPIAppConfigListeners) {
+                    listener.processConfigUpdate();
+                }
+            }
+        }
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        if (ProductInfo.getBetaEdition()) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(this, tc, "Deactivating OpenAPIAppConfigProviderImpl");
+            }
+            config = null;
+        }
+    }
+
+    private void betaFenceCheck() throws UnsupportedOperationException {
+        // Not running beta edition, throw exception
+        if (!ProductInfo.getBetaEdition()) {
+            throw new UnsupportedOperationException("This method is beta and is not available.");
+        } else {
+            // Running beta exception, issue message if we haven't already issued one for this class
+            if (!issuedBetaMessage) {
+                Tr.info(tc, "BETA: A beta method has been invoked for the class " + this.getClass().getName() + " for the first time.");
+                issuedBetaMessage = !issuedBetaMessage;
+            }
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    @Deprecated
+    public MpConfigServerConfigObject getConfiguration() {
+        betaFenceCheck();
+        return (config);
+    }
+
+    @Override
+    public void registerAppConfigListener(OpenAPIAppConfigListener listener) {
+        openAPIAppConfigListeners.add(listener);
+    }
+
+    @Override
+    public void unregisterAppConfigListener(OpenAPIAppConfigListener listener) {
+        openAPIAppConfigListeners.add(listener);
+    }
```

Comment:
```suggestion
    @Override
    public void unregisterAppConfigListener(OpenAPIAppConfigListener listener) {
        openAPIAppConfigListeners.remove(listener);
    }
```

In addition, `openAPIAppConfigListeners` needs synchronization, or to be a concurrent list

Reply:
or a copy on write list